sqlmesh/core/test/definition.py

```diff
@@ -94,29 +94,46 @@ def shortDescription(self) -> t.Optional[str]:
     def setUp(self) -> None:
         """Load all input tables"""
         for name, values in self.body.get("inputs", {}).items():
-            # Types specified in the test take precedence over the corresponding inferred ones
+            all_types_are_known = False
+            known_columns_to_types: t.Dict[str, exp.DataType] = {}
+
             model = self.models.get(name)
-            columns_to_types = {
-                **(model.columns_to_types if model and model.annotated else {}),
-                **values.get("columns", {}),
-            }
+            if model:
+                inferred_columns_to_types = model.columns_to_types or {}
+                known_columns_to_types = {
+                    c: t for c, t in inferred_columns_to_types.items() if type_is_known(t)
+                }
+                all_types_are_known = len(known_columns_to_types) == len(inferred_columns_to_types)
```

Comment:
This doesn't look right. If we couldn't infer the columns, the `model.columns_to_types` returns None and the length will be 0. Wouldn't we incorrectly conclude that all types are known then?

Reply:
```
all_types_are_known = inferred_columns_to_types  and len(known_columns_to_types) == len(inferred_columns_to_types)
```